scripts/scripts.js

```diff
@@ -99,6 +99,118 @@ function buildModalFragmentBlock(main) {
   }
 }
 
+function indexOfElementInParent(element) {
+  return [...element.parentElement.children].indexOf(element);
+}
+
+/**
+ * Split children of this div up into 1, 2 or 3 separate divs with cut points as specified in
+ * the from and to indexes, separating the elements from-to into
+ * a separate div on the same level and putting the remaining elements in new divs surrounding it.
+ * @param {HTMLElement} div The element to work on.
+ * @param {number} from The index from from which to put element into the middle div.
+ * @param {number} to The index up-to-but-not-including the element that marks then end of the
+ * middle div.
+ * @returns Returns the middle div.
+ */
+export function splitChildDiv(div, from, to) {
+  // run backwards because moving element will delete them from the original
+
+  let afterDiv;
+  if (to < div.children.length - 1) {
+    afterDiv = document.createElement('div');
+    for (let i = div.children.length - 1; i >= to; i -= 1) {
+      afterDiv.prepend(div.children[i]);
+    }
+  }
+
+  const midDiv = document.createElement('div');
+  for (let i = to - 1; i >= from; i -= 1) {
+    midDiv.prepend(div.children[i]);
+  }
+
+  let beforeDiv;
+  if (from > 0) {
+    beforeDiv = document.createElement('div');
+    for (let i = from - 1; i >= 0; i -= 1) {
+      beforeDiv.prepend(div.children[i]);
+    }
+  }
+
+  if (beforeDiv) {
+    div.parentElement.insertBefore(beforeDiv, div);
+  }
+  div.parentElement.insertBefore(midDiv, div);
+  if (afterDiv) {
+    div.parentElement.insertBefore(afterDiv, div);
+  }
+  div.parentElement.removeChild(div);
+
+  return midDiv;
+}
+
+function buildImageCollageForPicture(picture, caption, buildBlockFunction) {
+  const newBlock = buildBlockFunction('image-collage', { elems: [picture, caption] });
+  newBlock.classList.add('boxy-col-1');
+  return newBlock;
+}
+
+function buildImageWithCaptionForPicture(parentP, picture, buildBlockFunction) {
+  const enclosingDiv = parentP.parentElement;
+  if (enclosingDiv) {
+    // The caption could either be right next to the picture (if on the same line)
+    // or it could be in an adjacent sibling element (if 'enter' was pressed between)
+    const captionP = [
+      picture.nextElementSibling,
+      parentP.nextElementSibling,
+    ];
+
+    // eslint-disable-next-line no-restricted-syntax
+    for (const cp of captionP) {
+      if (!cp) {
+        // eslint-disable-next-line no-continue
+        continue;
+      }
+
+      if (cp.localName === 'em') {
+        // It's on the same line
+        enclosingDiv.append(buildImageCollageForPicture(picture, cp, buildBlockFunction));
+        return;
+      }
+
+      // Maybe the 'em' is on the next line, which means its in a separate <p> element
```

Comment:
I think the assumption here that it will be in a separate `<p>` element is what leads to what I said in my earlier comment

![Screenshot 2023-10-02 at 17 43 05](https://github.com/hlxsites/sunstar/assets/44490413/4b70952d-fd10-4b9a-a271-cca55e5f8b0d)

Reply:
Yes you can basically trigger the image-with-caption by either specifying the caption right after the image (on the same line) or in the next paragraph (by pressing enter). 
If you press shift-enter it doesn't do it and I think that's probably what you want...

The fact that one of them is smaller is just because one of them is the `image-collage` block and the other isn't

Reply:
I agree that we should have a mechanism to skip the autoblocking, but I'm not sure if the current approach is the best from an authoring perspective; the two cases look almost identical in the doc, and personally I only figured this out by accident

![Screenshot 2023-10-03 at 09 41 24](https://github.com/hlxsites/sunstar/assets/44490413/a39529bb-23ee-440f-87c0-9bc37d524860)

We could maybe only keep the caption right next to the image?

As for the picture getting smaller, I really don't like this, this page for example should have one captioned image and another one that does not have a caption, but they should both be the same size, which could not happen with the current implementation

https://main--sunstar--hlxsites.hlx.page/about/business-performance

Reply:
Isn't the shift-enter behaviour similar to elsewhere in word? If you have a bulleted list pressing enter will give you the next bullet but pressing shift-enter will suppress it?

On the size getting smaller - this is just the settings of the image-collage block. It's fine for the career pages use-case but maybe for your page you might need to tweak it?

Reply:
> Isn't the shift-enter behaviour similar to elsewhere in word? If you have a bulleted list pressing enter will give you the next bullet but pressing shift-enter will suppress it?

Yeah, ok, you're right